### DISCUSSION TITLE
how to check if string is a interger

### DISCUSSION THREAD

**Discussion by Kostia06** [vlang/v#15846] [General]
Hi, I am new to this programming language. I tried searching in vlang docs, trying to find how to check if string is a interger.
For example:
```
num := "50"
if num.is_int(){
    println("is int")
}else{
    println("is not int") 
}
```
it should print "is int"

**Comment by mnuccioarpae** [comment]
I'm also new to V language, but here is how I'm doing it until I discover a better way.

If you parse the string with `parse_int`, you get an error if it's not an integer value.
```V
num := "50"
inum := num.parse_int(10, 64) or {
    println(err)
    return
}
println("$num = $inum")
```

**Reply by Kostia06** [reply]
Thank you, but could you please tell me what are the arguments 10 and 64 are for?

**Reply by mnuccioarpae** [reply]
See https://modules.vlang.io/index.html#string.parse_int

10 is the base
64 the bits of the parsed int

**Reply by mnuccioarpae** [reply]
Here is a more complete example:
```V
num := "a50"
inum := num.parse_int(16, 64) or {
    println(err)
    return
}
println("$num = $inum")

if inum2 := num.parse_int(10, 64) {
    println("$num = $inum2")
} else {
    println("$num is not a base 10 integer")
}
```
